clickhouse/base/sslsocket.cpp

```diff
@@ -185,7 +185,7 @@ SSL_CTX * SSLContext::getContext() {
 // Allows caller to use returned value of `statement` if there was no error, throws exception otherwise.
 #define HANDLE_SSL_ERROR(SSL_PTR, statement) [&] { \
     if (const auto ret_code = (statement); ret_code <= 0) { \
-        throwSSLError(SSL_PTR, SSL_get_error(SSL_PTR, ret_code), LOCATION, #statement); \
+        throwSSLError(SSL_PTR, SSL_get_error(SSL_PTR, static_cast<int>(ret_code)), LOCATION, #statement); \
```

Comment:
We could've this piece of error checking code wrong from the very beginning, i.e. treating some non-error-code related return value as signed error code, or handle some API error codes wrongly (because it is long int, or unsigned int or any other reason).

Could you please post all related warning messages here in comments (or link to CI job that have those), so I can investigate it closely.

Reply:
![image](https://github.com/ClickHouse/clickhouse-cpp/assets/19810905/364bf53b-63eb-4e04-9543-15bf4546229c)

Reply:
Yeah, that is poor API design from OpenSSL's side, it should be safe to cast to int in this case.